src/nvapi/nvapi_d3d12_device.cpp

```diff
@@ -27,180 +91,86 @@ namespace dxvk {
-            log::info("Failed to find CuBIN in m_cubinSmemMap, defaulting to 0");
-
-        if (interfaceVersion >= 1)
-            return SUCCEEDED(cmdList->LaunchCubinShaderEx(reinterpret_cast<D3D12_CUBIN_DATA_HANDLE*>(pShader), blockX, blockY, blockZ, smem, params, paramSize, nullptr, 0));
-        else {
-            if (smem != 0)
-                log::info("Non-zero SMEM value supplied for CuBIN but ID3D12GraphicsCommandListExt1 not supported! This may cause corruption");
-
-            return SUCCEEDED(cmdList->LaunchCubinShader(reinterpret_cast<D3D12_CUBIN_DATA_HANDLE*>(pShader), blockX, blockY, blockZ, params, paramSize));
+        if (m_supportsExtGraphicsCommandList1 || log::tracing()) {
+            std::scoped_lock lock(m_cubinSmemMutex);
+            auto it = m_cubinSmemMap.find(pShader);
+            if (it != m_cubinSmemMap.end())
+                smem = it->second;
+            else
+                log::info("Failed to find CuBIN in m_cubinSmemMap, defaulting to 0");
         }
-    }
-
-    bool NvapiD3d12Device::CaptureUAVInfo(ID3D12Device* device, NVAPI_UAV_INFO* pUAVInfo) {
-        auto cubinDevice = GetCubinDevice(device);
-        if (cubinDevice == nullptr)
-            return false;
-
-        return SUCCEEDED(cubinDevice->CaptureUAVInfo(reinterpret_cast<D3D12_UAV_INFO*>(pUAVInfo)));
-    }
-
-    bool NvapiD3d12Device::IsFatbinPTXSupported(ID3D12Device* device) {
-        auto cubinDevice = GetCubinDevice(device);
-        return cubinDevice != nullptr;
-    }
-
-    // We are going to have single map for storing devices with extensions D3D12_VK_NVX_BINARY_IMPORT & D3D12_VK_NVX_IMAGE_VIEW_HANDLE.
-    // These are specific to NVIDIA and both of these extensions goes together.
-    Com<ID3D12DeviceExt> NvapiD3d12Device::GetCubinDevice(ID3D12Device* device) {
-        std::scoped_lock lock(m_cubinDeviceMutex);
-        auto it = m_cubinDeviceMap.find(device);
-        if (it != m_cubinDeviceMap.end())
-            return it->second;
-
-        auto cubinDevice = GetDeviceExt(device, D3D12_VK_NVX_BINARY_IMPORT);
-        if (cubinDevice != nullptr)
-            m_cubinDeviceMap.emplace(device, cubinDevice.ptr());
-
-        return cubinDevice;
-    }
 
-    Com<ID3D12DeviceExt> NvapiD3d12Device::GetDeviceExt(ID3D12Device* device, D3D12_VK_EXTENSION extension) {
-        Com<ID3D12DeviceExt> deviceExt;
-        if (FAILED(device->QueryInterface(IID_PPV_ARGS(&deviceExt))))
-            return nullptr;
+        if (m_supportsExtGraphicsCommandList1 >= 1)
+            return SUCCEEDED(m_vkd3dGraphicsCommandList->LaunchCubinShaderEx(reinterpret_cast<D3D12_CUBIN_DATA_HANDLE*>(pShader), blockX, blockY, blockZ, smem, params, paramSize, nullptr, 0));
 
-        if (!deviceExt->GetExtensionSupport(extension))
-            return nullptr;
+        if (log::tracing() && smem != 0)
+            log::info("Non-zero SMEM value supplied for CuBIN but ID3D12GraphicsCommandListExt1 not supported! This may cause corruption");
```

Comment:
This is probably serious enough that I'd log it even if tracing isn't enabled.

Reply:
I've restored the log message.